xc/xenops_server_xen.ml

```diff
@@ -1095,7 +1095,12 @@ module VM = struct
                   (domain_config, persistent)
               in
               let create_info = generate_create_info ~xc ~xs vm persistent in
-              let domid = Domain.make ~xc ~xs create_info domain_config (uuid_of_vm vm) in
+              let build_info =
+                match persistent.VmExtra.build_info with
```

Comment:
I think that for a VM start, the `build_info` will not be present here. It is currently constructed in `build_domain_exn`, just before calling `Domain.build`, and then stored in the `persistent` record in the local DB.

Reply:
If all you need is vcpus, then you can probably just get that from `vm.vcpu_max`.

Reply:
This whole build_info vs create_info was conceptually broken in `xend`, and remains so in libxl and Xapi.  In this case, all that matters is vm.vcpu_max, which appears to be passed into `build_domain_exn`, so comes from elsewhere.